From d562b2187263b40aacc1a50d3f25db2cf28696d6 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.org>
Date: Tue, 9 Jul 2019 10:32:40 +0100
Subject: [PATCH] overlays: i2c-gpio: Fix the "bus" parameter

The "bus" parameter has two functions - providing unique names for
multiple instances of the overlay, and allowing the number of the bus
(i.e. "i2c-<bus>") to be specified. The second function hasn't worked
as intended because the overlay doesn't include a "reg" property and
the firmware intentionally won't create a "reg" property if one doesn't
already exist.

Allow the bus numbering scheme to work as intended by providing a "reg"
with a default value that means "the next available one".

See: https://github.com/raspberrypi/linux/issues/3062

Signed-off-by: Phil Elwell <phil@raspberrypi.org>
---
 arch/arm/boot/dts/overlays/i2c-gpio-overlay.dts | 2 ++
 1 file changed, 2 insertions(+)

--- a/arch/arm/boot/dts/overlays/i2c-gpio-overlay.dts
+++ b/arch/arm/boot/dts/overlays/i2c-gpio-overlay.dts
@@ -7,8 +7,10 @@
 
 	fragment@0 {
 		target-path = "/";
+
 		__overlay__ {
 			i2c_gpio: i2c@0 {
+				reg = <0xffffffff>;
 				compatible = "i2c-gpio";
 				gpios = <&gpio 23 0 /* sda */
 					 &gpio 24 0 /* scl */
